scripts/py_matter_idl/matter/idl/tests/inputs/large_all_clusters_app.matter

```diff
@@ -4090,7 +4090,7 @@ endpoint 0 {
     callback attribute packetUnicastTxCount;
     callback attribute currentMaxRate;
     callback attribute overrunCount;
-    ram      attribute featureMap default = 3;
+    ram      attribute featureMap default = 0;
```

Comment:
Why is this being changed?

Reply:
Because in this matter file, WifiNetworkDiagnostics has no features: https://github.com/project-chip/connectedhomeip/blob/master/scripts/py_matter_idl/matter/idl/tests/inputs/large_all_clusters_app.matter#L1371

The config was saying "you have features 1 and 2" and there were no features like that defined.

Reply:
The Wi-Fi network diagnostics cluster has feature bits 0 and 1 defined, so 3 would correspond to both those bits being set...

And they're in the XML:

```
    <features>
      <feature bit="0" code="PKTCNT" name="PacketCounts" summary="Node makes available the counts for the number of received and transmitted packets on the ethernet interface.">
        <optionalConform/>
      </feature>
      <feature bit="1" code="ERRCNT" name="ErrorCounts" summary="Node makes available the counts for the number of errors that have occurred during the reception and transmission of packets on the ethernet interface.">
        <optionalConform/>
      </feature>
    </features>
```

So the question then is, I guess: why does this .matter file not have that information?

Reply:
The .matter file was generated before that XML was available (or maybe the structures are not selected?).

These tests use matter files as is, there is no xml loading to re-generate after zap updates. I think we tried hard fo that as we did not want to chase all-clusters updates (that update on every cluster change) since that creates a lot of merge conflicts.

We now have static files, which is good on merge conflicts but bad on "you expect something in the current spec/implementation, but the snapshot this works with is old".

Reply:
Specifically: these unit tests test "generaly from a matter file" and this matter file was a `copy the matter file from all clusters` at some point in time. There is no `.zap` to keep this always updated, so it is a fixed point in time and misses any XML updates we did.

When it was copied over the feature maps were wrong. We added those feature maps in the mean time in the spec, but that addition did not make it through.

Reply:
These feature maps have been in the spec for a good long while...  Sounds like we need to do a followup to make this .matter file sane?

Reply:
Maybe... updated to latest matter so this should be resolved.